craft_application/application.py

```diff
@@ -205,7 +214,15 @@ def run_managed(self, platform: str | None, build_for: str | None) -> None:
         build_plan = _filter_plan(build_plan, platform, build_for)
 
         for build_info in build_plan:
-            extra_args["env"] = {"CRAFT_PLATFORM": build_info.platform}
+            env = {"CRAFT_PLATFORM": build_info.platform}
+
+            if self.app.features.build_secrets:
+                # If using build secrets, put them in the environment of the managed
+                # instance.
+                secret_values = cast(secrets.BuildSecrets, self._secrets)
```

Comment:
Isn't `self._secrets` already `BuildSecrets`? (why is the cast necessary?)

Reply:
It's a `BuildSecrets | None` because the specific application might not be using build-secrets.

Reply:
So here's my thought process: I started with a `is not None` check, but then it occurred to me that this shouldn't happen: if the feature is enabled, `self._secrets` being `None` is an error. 
Then I *could* add a check for that here but to test it I would have to write a really contrived piece of test code, so I ended up leaving this cast here. If somehow `self._secrets` is None then we'll get an AttributeError.

... what I _could_ do, to remove the need for the cast, is declare the attribute as `self._secrets: BuildSecrets = None` with a type ignore in `__init__()`, and then the rest of the code can assume that the variable is set if the feature is. I don't have strong opinions either way so I can go with either.